applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -13,145 +13,88 @@ class FluidDynamicsAnalysis(AnalysisStage):
-        factory = KratosProcessFactory(self.model)
+    def _CreateProcesses(self, parameter_name, initialization_order):
+        """Create a list of Processes
+        This method is TEMPORARY to not break existing code
+        It will be removed in the future
+        """
+        list_of_processes = super(FluidDynamicsAnalysis, self)._CreateProcesses(parameter_name, initialization_order)
+
         # The list of processes will contain a list with each individual process already constructed (boundary conditions, initial conditions and gravity)
         # Note 1: gravity is constructed first. Outlet process might need its information.
         # Note 2: initial conditions are constructed before BCs. Otherwise, they may overwrite the BCs information.
-        self.list_of_processes =  factory.ConstructListOfProcesses( self.project_parameters["gravity"] )
-        self.list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["initial_conditions_process_list"] )
-        self.list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["boundary_conditions_process_list"] )
-        self.list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["auxiliar_process_list"] )
-
-        #TODO this should be generic
-        # initialize GiD  I/O
-        self.output = self._SetUpGiDOutput()
-        if self.output is not None:
-            self.list_of_processes += [self.output,]
-
-    def _SetUpAnalysis(self):
-        '''
-        Initialize the Python solver and its auxiliary tools and processes.
-        This function should prepare everything so that the simulation
-        can start immediately after exiting it.
-        '''
-
-        for process in self.list_of_processes:
-            process.ExecuteInitialize()
-
-        self.solver.Initialize()
-
-        ## If the echo level is high enough, print the complete list of settings used to run the simualtion
-        if self.is_printing_rank and self.echo_level > 1:
-            with open("ProjectParametersOutput.json", 'w') as parameter_output_file:
-                parameter_output_file.write(self.project_parameters.PrettyPrintJsonString())
-
-        ## Stepping and time settings
-        self.end_time = self.project_parameters["problem_data"]["end_time"].GetDouble()
-
-        if self.main_model_part.ProcessInfo[Kratos.IS_RESTARTED]:
-            self.time = self.main_model_part.ProcessInfo[Kratos.TIME]
+        if parameter_name == "processes":
+            old_processes_names = ["gravity", "initial_conditions_process_list", "boundary_conditions_process_list", "auxiliar_process_list"]
+            if len(list_of_processes) == 0:
+                KratosMultiphysics.Logger.PrintInfo("FluidDynamicsAnalysis", "Using the old way to create the processes, this will be removed!")
+                from process_factory import KratosProcessFactory
+                factory = KratosProcessFactory(self.model)
```

Comment:
could u explain in short what is happening here?

the KratosProcessFactory is still the same right? or did u change it?

Reply:
I did not change the ProccesFactory
If I enter this `if` it means that there are no processes inside a `processes` dict present in the ProjectParams. This is how the current interface looks like

Then I construct the processes the same way they were constructed previously